### PR TITLE
Upgrade to k2, remove Kotlin/js

### PR DESCRIPTION
**Pull request by Daoortor**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -78,19 +78,20 @@ configure(subprojects.filter { it.name != "common" && frontendSuffix !in it.name
         implementation("org.jetbrains.academy.test.system:kotlin-test-system:1.0.9")
     }
 
-    val jvmVersion = "11"
+    val jvmVersion = JvmTarget.JVM_11
+    val jvmVersionNumber = jvmVersion.toString().takeLastWhile { it.isDigit() }
```
Comment (mikrise2) [inline build.gradle.kts:82]: Please rename it and also just use `jvmVersion.target`

File: .github/workflows/qodana.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+          fetch-depth: 0  # a full history is required for pull request analysis
+
+      - name: 'Qodana Scan'
+
+        uses: JetBrains/qodana-action@v2024.1
+
+        env:
+
+          QODANA_TOKEN: ${{ secrets.QODANA_TOKEN }}
```
Comment (mikrise2) [inline .github/workflows/qodana.yml:53]: I think here is some problems with the code formatting

**Review by mikrise2 — CHANGES_REQUESTED** [review]
Good job!
Please also fix workflows files as I wrote [here](https://github.com/jetbrains-academy/kotlin-onboarding-object-oriented-programming/pull/172)

**Comment by Daoortor** [comment]
Apparently, Qodana build fails now because the QODANA_TOKEN secret is not in the head repository.

File: wordsGeneratorServer/wordsGeneratorServerFinishGame/src/main/kotlin/jetbrains/kotlin/course/words/generator/results/GameResultsService.kt [hunk 3]
```diff
@@ -18,5 +18,6 @@ class GameResultsService {
         gameHistory.add(result)
     }
 
-    fun getAllGameResults() = gameHistory.reversed()
+    // It's necessary to specify return type explicitly, so kotlin-reflect gets it right
+    fun getAllGameResults(): List<GameResult> = gameHistory.reversed()
```
Comment (nbirillo) [inline wordsGeneratorServer/wordsGeneratorServerFinishGame/src/main/kotlin/jetbrains/kotlin/course/words/generator/results/GameResultsService.kt:22]: What if the student does not explicitly specify the return type? I'd suggest adding a check that will handle this case in the tests, WDYT?

**Review by nbirillo — CHANGES_REQUESTED** [review]
As I can see you have not updated the resources in the servers with updated compiled front end, please do it to double check if everything is working. You need to build the frontend (yarn run build) and then run the `serveResourcesTask` task for the necessary project.

I will come back soon with a token for Qodana

**Pull request merged**